test/mercury/baseuriparameters.test.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+/* eslint-disable no-undef */
+"use strict";
+import { validateFile } from "../../src/validator";
+import {
+  getHappySpec,
+  conforms,
+  breaksOnlyOneRule,
+  renderSpecAsFile
+} from "../utils.test";
+
+const PROFILE = "mercury";
+
+describe("baseUriParameters checking tests", () => {
+  it("conforms when shortCode is only baseUriParameter", async () => {
+    const doc = getHappySpec();
+    const result = await validateFile(renderSpecAsFile(doc), PROFILE);
+    conforms(result);
+  });
+
+  it("does not conform when shortCode is missing from baseUriParameters", async () => {
+    const doc = getHappySpec();
+    // We have to change the baseUri because it's implicit if it appears in the template
+    doc["baseUri"] = "https://example.com";
+    delete doc["baseUriParameters"]["shortCode"];
+    const result = await validateFile(renderSpecAsFile(doc), PROFILE);
+    breaksOnlyOneRule(
+      result,
+      "http://a.ml/vocabularies/data#baseuriparameters-must-only-have-shortcode"
+    );
+  });
+
```

Comment:
Pls add a test for url like "https://{shortCode}.api.commercecloud.salesforce.com/{anotherParameterInPlaceOfApiFamily}/shopper-orders/{version}/";

Reply:
done